extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -100,8 +112,19 @@ export class AmpAdExit extends AMP.BaseElement {
     if (target.vars) {
       for (const customVar in target.vars) {
         if (customVar[0] == '_') {
-          vars[customVar] = () =>
-              args[customVar] || target.vars[customVar].defaultValue;
+          const vals =
+              /** @type {VariableFrom3pAnalytics} */ (target.vars[customVar]);
+          vars[customVar] = () => {
+            if (vals.vendorAnalyticsSource) {
+              const map = IframeTransportResponseMap.get(this.getAmpDoc(),
+                  vals.vendorAnalyticsSource,
+                  /** @type {!string} */ (this.win.document.baseURI));
```

Comment:
the string type is non-null by default, so ! is redundant

https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System